siliconcompiler/core.py

```diff
@@ -990,18 +989,30 @@ def _search(self, cfg, keypath, *args, field='value', mode='get', clobber=True):
                             else:
                                 self.logger.error(f'{field} must be set to boolean.')
                                 self.error = 1
-                        elif field in ('filehash', 'date', 'author', 'signature'):
-                            if isinstance(val, list):
+                        elif field in ('hashalgo', 'scope', 'require','type',
+                                       'shorthelp', 'switch', 'help'):
```

Comment:
Could these small sets of type-convertible parameters be stored somewhere in `schema.py`, or are they unlikely to be reused?

Reply:
I thought about it, but there are some things that are simply too coupled to the set/get routine. Let's leave it like this.